sqlcapture/capture.go

```diff
@@ -194,11 +201,27 @@ func (c *Capture) Run(ctx context.Context) (err error) {
 		return fmt.Errorf("error streaming until watermark: %w", err)
 	}
 	for _, streamID := range c.State.StreamsInState(TableModePending) {
-		logrus.WithField("stream", streamID).Info("activating replication for stream")
+		var binding = c.Bindings[streamID]
+		if binding == nil {
+			return fmt.Errorf("internal error: no binding information for stream %q", streamID)
+		}
+
+		logrus.WithFields(logrus.Fields{"stream": streamID, "mode": binding.Resource.Mode}).Info("activating replication for stream")
+
 		var state = c.State.Streams[streamID]
-		state.Mode = TableModeBackfill
+		switch binding.Resource.Mode {
+		case BackfillModeNormal, "":
```

Comment:
Unless I'm missing something, I'd tend to prefer having `(r Resource) Validate()` set this value to `"Normal"` if it's an empty string. That would make this case check more direct. But it doesn't look like we are doing an empty string check anywhere other than here, so I don't feel super strongly about this.

Reply:
Yeah, making that get defaulted somewhere earlier in startup seems reasonable. I think we generally prefer to have `Validate()` be side-effect free (see for instance the `Validate()` vs `SetDefault()` methods on the connector config structs) so I'll probably do it that way though.